.github/workflows/create-release.yml

```diff
@@ -1,279 +1,456 @@
+          current_run_nodejs_16_sha256_image_hash_code=$(docker image inspect registry.access.redhat.com/ubi8/nodejs-16-minimal --format '{{json .RepoDigests}}' | jq -r '.[0]')
+          previous_run_nodejs_16_sha256_image_hash_code=$(cat ${{ github.workspace }}/${{ env.UBI_RUN_NODEJS_16_SHA256_FILENAME }})
+
+          if [ "$previous_run_nodejs_16_sha256_image_hash_code" != "$current_run_nodejs_16_sha256_image_hash_code" ]; then
+            ubi_images_need_update+=("node-16-run")
+          fi
+
+          ubi_images_need_update+=("node-16-run")
+
+          # Run Image Node.js 18
+          docker pull registry.access.redhat.com/ubi8/nodejs-18-minimal
+          current_run_nodejs_18_sha256_image_hash_code=$(docker image inspect registry.access.redhat.com/ubi8/nodejs-18-minimal --format '{{json .RepoDigests}}' | jq -r '.[0]')
+          previous_run_nodejs_18_sha256_image_hash_code=$(cat ${{ github.workspace }}/${{ env.UBI_RUN_NODEJS_16_SHA256_FILENAME }})
+
+          if [ "$previous_run_nodejs_18_sha256_image_hash_code" != "$current_run_nodejs_18_sha256_image_hash_code" ]; then
+            ubi_images_need_update+=("node-18-run")
+          fi
 
   create_stack:
     name: Create Stack
+    needs: poll_ubi_images
+    if: ${{ ! ( needs.poll_ubi_images.outputs.ubi_images_need_update == '[]' && github.event_name == 'schedule' ) }}
+
     runs-on: ubuntu-22.04
     steps:
-    - name: Checkout
-      uses: actions/checkout@v3
-
-    - name: Create stack
-      id: create-stack
-      uses: paketo-buildpacks/github-config/actions/stack/create-stack@main
-
-    - name: Upload build image
-      uses: actions/upload-artifact@v3
-      with:
-        name: current-build-image
-        path: build/build.oci
-
-    - name: Upload base run image
-      uses: actions/upload-artifact@v3
-      with:
-        name: current-run-image
-        path: build/run.oci
-
-    - name: Upload nodejs-16 run image
-      uses: actions/upload-artifact@v3
-      with:
-        name: current-nodejs-16-run-image
-        path: build-nodejs-16/run.oci
-
-    - name: Upload nodejs-18 run image
-      uses: actions/upload-artifact@v3
-      with:
-        name: current-nodejs-18-run-image
-        path: build-nodejs-18/run.oci
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Create stack
+        id: create-stack
+        uses: paketo-buildpacks/github-config/actions/stack/create-stack@main
+
+      - name: Upload nodejs 16 run image
+        uses: actions/upload-artifact@v3
+        with:
+          name: current-nodejs-16-run-image
+          path: ./build-nodejs-16/run.oci
+
+      ## build image
+      - name: Upload build image
+        uses: actions/upload-artifact@v3
+        with:
+          name: current-build-image
+          path: ./build/build.oci
+
+      - name: Get build image hash code
+        run: |
+          mkdir -p hash-codes/build
+          # improvement: we dont have to re-download the image to get the hash code,
```

Comment:
agreed we should avoid a second download.

Reply:
The implementation is here https://github.com/paketo-community/ubi-base-stack/pull/7/commits/6ee311aa86a44f1c336b7b5ef779bbcdf87b972c